### ISSUE TITLE
0.67.7 Regression when in rail mode - content left margin

### ISSUE THREAD

**Issue by Sheepux** [carbon-design-system/carbon-components-svelte#1459]
#1428 Removed the left margin (that's fine) on content when sidenav is collapsed, but removes it completly when the sidenav is in rail mode.

Before  : the content was on the right of the sidenav rail
Now : now it's under the sidenav rail (z-index of the sidenav)
Expected: the content is on the right of the sidenav. Margin-left should be the size of the sidenav rail so that the extra part of the sidenav is over the content (z-index).

I'm guessing that -isSidenavRail- will have to be in the store as well so that -content- can add that extra margin again

**Comment by metonym** [comment]
Thanks for reporting – can you provide a repro to assist me in debugging?

I'm starting with the [Rail example](https://carbon-components-svelte.onrender.com/framed/UIShell/HeaderNavRail?theme=g10), but when I remove the `margin-left` override, it still looks the same.

The `marign-left: 3rem` rule is only applied to the UI Shell if `.bx--content` is a sibling of `header`. In the Rail example, this isn't the case.

**Comment by Sheepux** [comment]
I've pinpointed the behavior change in 0.67.6->0.67.7 (see enclosed image, 0.67.6 is on top)
![carbon0 67 7](https://user-images.githubusercontent.com/19878135/185999323-b6217173-ce1b-44f5-8d33-e21241a075e9.png)
Behavior can be seen in an automatic test page: https://haruneko.pages.dev/ 
I'll try to set up a repl with a bare project

**Comment by Sheepux** [comment]
here you go, simply change the carbon component version
https://svelte.dev/repl/37dcec10caaf4254961a43e528d3da89

Note: i've changed
```
<Content>
  <Grid>
    <Row>
      <Column>
        <h1>Welcome</h1>
      </Column>
    </Row>
  </Grid>
</Content>
```
to
```
<Content>
        <h1>Welcome</h1>
</Content>
```

**Comment by metonym** [comment]
Excellent – thank you.

**Comment by metonym** [comment]
Fixed in [v0.70.1](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.70.1).

I've also [updated](https://github.com/carbon-design-system/carbon-components-svelte/commit/6b515b7589cd901d5adb303517ca5c2cb232ba86) the UI Shell rail example in the docs to make the `SideNav` a sibling component to `Content`.

**Comment by Sheepux** [comment]
I hereby confirm that the issue is resolved, many thanks for the quick update :)
